components/form/builder/src/Input/index.js

```diff
@@ -0,0 +1,139 @@
+import React, {useCallback} from 'react'
+
+import PropTypes from 'prop-types'
+import {field} from '../prop-types'
+
+import MoleculeInputField from '@s-ui/react-molecule-input-field'
+
+import {FIELDS, DISPLAYS} from '../Standard'
+
+// mapping between DSL standard and HTML input type attribute
+const DISPLAY = {
+  [DISPLAYS[FIELDS.TEXT].EMAIL]: 'email',
+  [DISPLAYS[FIELDS.TEXT].PHONE]: 'tel',
+  [DISPLAYS[FIELDS.TEXT].TEXT]: 'text',
+  [DISPLAYS[FIELDS.TEXT].DEFAULT]: 'text'
+}
+
+const Input = ({
+  input,
+  tabIndex,
+  onChange,
+  size,
+  leftAddon,
+  rightAddon,
+  errors
+}) => {
+  const errorMessages = errors[input.id]
+  const onChangeCallback = useCallback(
+    evt => {
+      return onChange(input.id, evt.target.value)
+    },
+    [input, onChange]
+  )
+
+  let nextProps = {}
+  switch (input.type) {
+    case FIELDS.TEXT: {
+      nextProps = {type: DISPLAY[input.display] || 'text'} // this is the html input type (not the field type related to DSL form-builder)
+
+      const constraints = input.constraints || []
+
+      nextProps = constraints.reduce((acc, constraint) => {
```

Comment:
This kind of reduce and way of creating the object is very hard to read.

You might could consider something like this.

```
nextProps = constraints.reduce((acc, constraint) => {
  const {maxlength: maxLength, minlength: minLength, notnull, pattern} = constraint?.property?

  return {
    ...acc,
    ...(maxLength && { maxLength }),
    ...(minLength && { minLength }),
    ...(pattern && { pattern }),
    ...(notnull === '' && {
      required: true
    }),
  }
})
```

Reply:
Also, it should benefit the codebase, in order to avoid doing more than once the optional chaining transformation.

Reply:
I am going to merge and evaluate these optimizations in another PR.